agent/ecscni/plugin.go

```diff
@@ -100,13 +114,17 @@ func (client *cniClient) CleanupNS(
 
 // cleanupNS is called by CleanupNS to cleanup the task namespace by invoking DEL for given CNI configurations
 func (client *cniClient) cleanupNS(ctx context.Context, cfg *Config) error {
+	client.guard.lock()
+	defer client.guard.unlock()
+
 	seelog.Debugf("[ECSCNI] Cleaning up the container namespace %s", cfg.ContainerID)
 
 	runtimeConfig := libcni.RuntimeConf{
 		ContainerID: cfg.ContainerID,
 		NetNS:       cfg.ContainerNetNS,
 	}
 
+	var delError error = nil
```

Comment:
nit: Why is this explicitly set to `nil`?

Reply:
This is the initial assignment of the return variable.
I would prefer to be explicit about the value being returned unless it is being changed due to error.

Reply:
`var delError error` would also result in `delError` being `nil`

Reply:
Yes it would but what's the issue in the overt initialization of the variable?

Reply:
With golang, it is common practice to just have it as `var delError error`